enclave-runtime/src/lib.rs

```diff
@@ -905,54 +900,31 @@ where
 		debug!("Got following trusted calls from pool: {:?}", trusted_calls);
 	}
 
-	let mut calls = Vec::<OpaqueCall>::new();
-	let mut call_hashes = Vec::<H256>::new();
-
-	// load state before executing any calls
-	let (mut sidechain_db, state_lock) = state_handler
-		.load_for_mutation(&shard)
-		.map(|(l, s)| (SidechainDB::<SB::Block, _>::new(s), l))?;
-
-	let prev_state_hash = sidechain_db.state_hash();
-	trace!("state apriori hash: {:?}", prev_state_hash);
-
-	// update state needed for pallets
-	sidechain_db.set_block_number(&sidechain_db.get_block_number().map_or(1, |n| n + 1));
-	sidechain_db.set_timestamp(&now_as_u64());
+	let execution_result = stf_executor.execute_timed_calls_batch::<PB, _>(
+		&trusted_calls,
+		latest_onchain_header,
+		&shard,
+		max_exec_duration,
+		|s| {
+			let mut sidechain_db = SidechainDB::<SB::Block, _>::new(s);
+			sidechain_db.set_block_number(&sidechain_db.get_block_number().map_or(1, |n| n + 1));
+			sidechain_db.set_timestamp(&now_as_u64());
+			sidechain_db.ext
+		},
```

Comment:
The logic for setting block number, timestamp etc is now packed into a state 'pre-processing' lambda.

Reply:
I think, this looks very nice.